documentation/design-docs/dotnet-monitor.yaml

```diff
@@ -0,0 +1,473 @@
+schemes:
+- "https"
+- "http"
+paths:
+  /processes:
+    get:
+      tags:
+      - "processes"
+      operationId: "getprocesses"
+      produces:
+      - "application/json"
+      responses:
+        200:
+          description: "Success"
+          schema:
+            type: array
+            items: 
+              $ref: "#/definitions/Process"
+        400:
+          description: "Bad request"
+          schema:
+            $ref: "#/definitions/Error"
+  /dump:
+    get:
+      tags:
+      - "dumps"
+      operationId: "getdefaultprocessdump"
+      produces:
+      - "application/octet-stream"
+      parameters:
+      - name: "type"
+        in: query
+        description: "Specifies the dump type, which determines the kinds of information that are collected from the process."
+        type: string
+        enum:
+        - "Normal"
+        - "WithHeap"
+        - "Triage"
+        - "Full"
+        required: false
+        default: "Normal"
+      responses:
+        200:
+          description: "Success"
+          schema:
+            format: "byte"
+        400:
+          description: "Bad request"
+          schema:
+            $ref: "#/definitions/Error"
+  /dump/{uid}:
+    get:
+      tags:
+      - "dumps"
+      operationId: "getprocessdump"
+      produces:
+      - "application/octet-stream"
+      parameters:
+      - name: "uid"
+        in: "path"
+        description: "Process ID of the process"
+        required: true
+        type: "string"
+      - name: "type"
+        in: query
+        description: "Specifies the dump type, which determines the kinds of information that are collected from the process."
+        type: string
+        enum:
+        - "Normal"
+        - "WithHeap"
+        - "Triage"
+        - "Full"
+        required: false
+        default: "Normal"
+      responses:
+        200:
+          description: "Success"
+          schema:
+            format: "byte"
+        400:
+          description: "Bad request"
+          schema:
+            $ref: "#/definitions/Error"
+  /gcdump:
+    get:
+      tags:
+      - "dumps"
+      operationId: "getdefaultgcdump"
+      produces:
+      - "application/octet-stream"
+      parameters:
+      - name: "timeout"
+        in: "query"
+        description: "Maximum duration of the GC Dump"
+        required: false
+        type: "integer"
+        default: "30"
+      responses:
+        200:
+          description: "Success"
+          schema:
+            format: "byte"
+        400:
+          description: "Bad request"
+          schema:
+            $ref: "#/definitions/Error"
+  /gcdump/{uid}:
+    get:
+      tags:
+      - "dumps"
+      operationId: "getgcdump"
+      produces:
+      - "application/octet-stream"
+      parameters:
+      - name: "timeout"
+        in: "query"
+        description: "Maximum duration of the GC Dump"
+        required: false
+        type: "integer"
+        default: "30"
+      - name: "uid"
```

Comment:
uid?

Reply:
uid = pid | runtime_cookie

Reply:
Our naming gets confusing if we define that uid is pid | runtime_cookie and the json returned by the 'processes' REST call has a property called uid which has to return a particular value. I'd suggest uid in the process definition should be renamed 'runtime_cookie' and a separate 'pid' property should be added. This ensures that there is a distinct correct value for each of those properties to report. If we don't know what they are we can leave them blank.

We may also want to rename uid parameter in these REST calls as -> 'pid_or_runtime_cookie' for clarity?